### ISSUE TITLE
is gnumpy supported?

### ISSUE THREAD

**Issue by vzhong** [HIPS/autograd#37]
I read on the tutorial that GPU support is on the roadmap. However there are also traces of GPU specific code which refer to [HIPS/gpu_numpy](https://github.com/HIPS/gpu_numpy). Is autograd usable with gpu_numpy as is?

Loving autograd so far! I've been playing around with porting a neural net library from Theano to autograd and it has been going extremely smoothly :+1:

**Comment by duvenaud** [comment]
Good question!  The short answer is: yes, but gnumpy supports such a small fraction of Numpy commands that it's probably not worth using yet.  gpu_numpy is our unfinished attempt to clean up gnumpy and turn it into a full-featured GPU backend for Numpy.  We managed to get matrix multiplication working transparently, but then realized how much work it would take to get all the corner cases right for so many Numpy functions that we stopped working on it for now.

We added code to autograd to hook into a GPU backend, but as far as we know, there isn't a mature one yet.  I think there are several ongoing projects - the most promising one appears to be CuPy:
https://github.com/pfnet/chainer/blob/cupy/cupy/__init__.py

Any suggestions on which GPU backends to target are welcome!
